MediaBrowser.Controller/MediaEncoding/EncodingHelper.cs

```diff
@@ -110,16 +112,40 @@ private bool IsCudaSupported()
         {
             return _mediaEncoder.SupportsHwaccel("cuda")
                    && _mediaEncoder.SupportsFilter("scale_cuda", null)
-                   && _mediaEncoder.SupportsFilter("yadif_cuda", null);
+                   && _mediaEncoder.SupportsFilter("yadif_cuda", null)
+                   && _mediaEncoder.SupportsFilter("hwupload_cuda", null);
         }
 
-        private bool IsTonemappingSupported(EncodingJobInfo state, EncodingOptions options)
+        private bool IsOpenclTonemappingSupported(EncodingJobInfo state, EncodingOptions options)
         {
             var videoStream = state.VideoStream;
+            if (videoStream == null)
+            {
+                return false;
+            }
+
             return IsColorDepth10(state)
                    && _mediaEncoder.SupportsHwaccel("opencl")
+                   && _mediaEncoder.SupportsFilter("tonemap_opencl", null)
                    && options.EnableTonemapping
-                   && string.Equals(videoStream.VideoRange, "HDR", StringComparison.OrdinalIgnoreCase);
+                   && (string.Equals(videoStream.ColorTransfer, "smpte2084", StringComparison.OrdinalIgnoreCase)
+                       || string.Equals(videoStream.ColorTransfer, "arib-std-b67", StringComparison.OrdinalIgnoreCase));
+        }
+
+        private bool IsCudaTonemappingSupported(EncodingJobInfo state, EncodingOptions options)
+        {
+            var videoStream = state.VideoStream;
+            if (videoStream == null)
+            {
+                return false;
+            }
+
+            return IsColorDepth10(state)
+                   && _mediaEncoder.SupportsHwaccel("cuda")
+                   && _mediaEncoder.SupportsFilter("tonemap_cuda", "GPU accelerated HDR to SDR tonemapping")
```

Comment:
This is out of the scope of this PR, but I checked `SupportsFilter` and it calls ffmpeg. We should really cache the result somewhere. Calling it for every transcode is a bit much.